src/client/options/mod.rs

```diff
@@ -192,6 +192,63 @@ impl fmt::Display for StreamAddress {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[non_exhaustive]
+pub enum ServerApiVersion {
+    Version1,
+}
+
+impl ServerApiVersion {
+    pub fn from_string (version: &str) -> Self {
+        match version {
+            "1" => Self::Version1,
```

Comment:
I think the most idiomatic way to do this would be to implement the [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) and [`Display`](https://doc.rust-lang.org/std/fmt/trait.Display.html) traits.

Reply:
Changed the enum to implement both `FromStr` and `Display`.